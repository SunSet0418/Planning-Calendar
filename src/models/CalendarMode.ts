export enum CalendarMode {
    Month = '월',
    Week = '주'
}