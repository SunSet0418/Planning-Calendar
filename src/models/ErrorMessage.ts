export class ErrorMessage {
    constructor(
        public title: string,
        public description: string
    ) { }
}