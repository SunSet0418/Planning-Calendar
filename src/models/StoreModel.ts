import { TaskStoreModel } from '../modules/reducers/task';

export interface StoreModel {
    task: TaskStoreModel
}