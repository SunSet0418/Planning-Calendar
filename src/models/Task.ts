export interface Task {
    id: number
    startDate: string,
    endDate: string,
    startTime: number,
    durationTime: number
}