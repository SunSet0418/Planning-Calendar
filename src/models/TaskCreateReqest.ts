export interface TaskCreateRequest {
    name: string,
    date: string,
    startHour: number,
    endHour: number
}