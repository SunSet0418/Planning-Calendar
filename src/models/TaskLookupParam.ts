export class TaskLookupParam {
    constructor(
        public year: number,
        public month: number
    ) {}
}