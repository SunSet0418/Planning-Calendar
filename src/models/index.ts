export { TaskStore } from './TaskStore';
export { Task } from './Task';
export { StoreModel } from './StoreModel';
export { CalendarMode } from './CalendarMode';
export { ErrorMessage } from './ErrorMessage';