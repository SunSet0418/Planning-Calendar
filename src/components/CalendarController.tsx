import * as React from 'react';
import styled from 'styled-components';
import { CalendarMode } from '../models/CalendarMode';
import { Moment } from 'moment';
import { weekOfMonth } from '../libs/weekOfMount';
import { Button } from 'antd';
import { CalendarModeDropdown } from './CalendarModeDropdown';
import 'antd/lib/button/style/css';

const Container = styled.div`
    padding: 20px;
`
const Label = styled.h3`
    display: inline;
`
Label.displayName = 'Label';

interface OwnProps {
    currentMoment: Moment,
    mode: CalendarMode,
    onChangeMode: (mode: CalendarMode) => void,
    onNext: () => void,
    onPrev: () => void
}

type Props = OwnProps & React.HTMLAttributes<HTMLDivElement>;

export const CalendarController: React.FC<Props> = (props: Props) => {
    const { currentMoment, mode, onChangeMode, onNext, onPrev, ...divProps } = props;

    let label = `${currentMoment.year()}년 / ${currentMoment.month() + 1}월`;
    if (mode === CalendarMode.Week) {
        label = `${label} / ${weekOfMonth(currentMoment)}주`;
    }

    return (
        <Container {...divProps}>
            <Button onClick={onPrev} shape="circle" icon="left" />
            <Button onClick={onNext} shape="circle" icon="right" />
            <Label>{label}</Label>
            <CalendarModeDropdown onChangeMode={onChangeMode} currentMode={mode} />
        </Container>
    )
}