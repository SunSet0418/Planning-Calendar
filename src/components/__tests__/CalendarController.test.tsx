import * as React from 'react';
import * as Enzyme from 'enzyme';
import { CalendarController } from '../CalendarController';
import { CalendarMode } from '../../models/CalendarMode';
import * as moment from 'moment';
import console = require('console');

describe('<CalendarController />', () => {
    const nextMock = jest.fn();
    const prevMock = jest.fn();
    const now = moment('2014-02-27T10:00:00');
    let component = null;

    it('renders correctly', () => {
        component = Enzyme.shallow(<CalendarController currentDate={now} mode={CalendarMode.Month} onNext={nextMock} onPrev={prevMock} />);
    });

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should 2014년 / 2월', () => {
        expect(component.find('h3').at(0).text()).toBe('2014년 / 2월');
    });

});