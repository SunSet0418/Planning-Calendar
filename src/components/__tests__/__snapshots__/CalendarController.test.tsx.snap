// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarController /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentMoment={"2014-02-27T10:00:00.000Z"}
    mode="월"
    onChangeMode={[Function]}
    onChangeMoment={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          icon="left"
          loading={false}
          onClick={[Function]}
          shape="circle"
/>,
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          icon="right"
          loading={false}
          onClick={[Function]}
          shape="circle"
/>,
        <Unknown>
          2014년 / 2월
</Unknown>,
        <Unknown
          currentMode="월"
          onChangeMode={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "ghost": false,
          "htmlType": "button",
          "icon": "left",
          "loading": false,
          "onClick": [Function],
          "shape": "circle",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "ghost": false,
          "htmlType": "button",
          "icon": "right",
          "loading": false,
          "onClick": [Function],
          "shape": "circle",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "2014년 / 2월",
        },
        "ref": null,
        "rendered": "2014년 / 2월",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    display: inline;
",
            ],
          },
          "displayName": "Label",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "h3",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentMode": "월",
          "onChangeMode": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    padding: 20px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            block={false}
            ghost={false}
            htmlType="button"
            icon="left"
            loading={false}
            onClick={[Function]}
            shape="circle"
/>,
          <Button
            block={false}
            ghost={false}
            htmlType="button"
            icon="right"
            loading={false}
            onClick={[Function]}
            shape="circle"
/>,
          <Unknown>
            2014년 / 2월
</Unknown>,
          <Unknown
            currentMode="월"
            onChangeMode={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "htmlType": "button",
            "icon": "left",
            "loading": false,
            "onClick": [Function],
            "shape": "circle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "ghost": false,
            "htmlType": "button",
            "icon": "right",
            "loading": false,
            "onClick": [Function],
            "shape": "circle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "2014년 / 2월",
          },
          "ref": null,
          "rendered": "2014년 / 2월",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    display: inline;
",
              ],
            },
            "displayName": "Label",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "h3",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentMode": "월",
            "onChangeMode": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    padding: 20px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
